Ch2 fr nadia (#31)
animint/animint-manual-fr: Chapitres/Ch02/Ch02-ggplot2.Rmd

knitr::opts_chunk$set(fig.path="Ch02-figures/")
```

Dans ce chapitre nous expliquons la grammaire des graphiques, un modèle puissant pour décrire une vaste gamme de visualisations de données.
<!-- comment -->
Après avoir lu ce chapitre, vous serez en mesure de
<!-- comment -->

> * Citer les avantages de la grammaire des graphiques par rapport aux systèmes précédents pour la visualisation de données

Reviewer: ```suggestion
* De citer les avantages de la grammaire des graphiques par rapport aux systèmes précédents de visualisation de données
```

<!-- comment -->

> * Installer le package R `animint2 `

Reviewer: ```suggestion
*D'Installer le package R `animint2 `
```

Author: ```suggestion
* D'Installer le package R `animint2`
```
merci Nadia

<!-- comment -->

> * Traduire les esquisses de graphique en code ggplot dans R

Reviewer: ```suggestion
* De traduire les esquisses de graphique en code ggplot dans R
```

<!-- comment -->
* Afficher des ggplots sur des pages web en utilisant `animint2`
<!-- comment -->

> * Créer des ggplots avec multiples couches

Reviewer: ```suggestion
* De créer des ggplots avec des couches multiples
```
```suggestion
* De créer des ggplots multi-couches
```

<!-- comment -->
* Créer des ggplots avec multiples panneaux

<!-- comment -->

## Historique et objectif de la grammaire des graphiques {#history}

<!-- comment -->

> La plupart des systèmes informatiques d'analyse de données proposent des fonctions permettant de créer des graphiques afin de visualiser les tendances dans les données.

Reviewer: ```suggestion
La plupart des systèmes informatiques d'analyse de données offrent des fonctions de création de graphiques pour visualiser les tendances des données.
```

<!-- comment -->
Les systèmes les plus anciens proposent des fonctions très générales permettant de tracer les éléments de base d'un graphique, tels que des lignes et des points (par exemple, les packages `graphics`  et `grid`  dans R).
<!-- comment -->
Si vous utilisez l'un de ces systèmes généraux, c'est à vous d'assembler les composants pour former un graphique significatif et interprétable.
<!-- comment -->

> L'avantage des systèmes généraux est qu'ils imposent peu de limites aux types de graphiques qui peuvent être créés.

Reviewer: ```suggestion
L'avantage des systèmes généraux réside dans le peu de limites imposées aux types de graphiques qui peuvent être créés.
```

<!-- comment -->

> L'inconvénient est que les systèmes généraux ne fournissent généralement pas de fonctions permettant d'automatiser les tâches courantes (axes, panneaux, légendes).

Reviewer: ```suggestion
Leur principal inconvénient est qu'ils  ne fournissent généralement pas de fonctions permettant d'automatiser les tâches courantes (axes, panneaux, légendes).
```

<!-- comment -->

Pour remédier aux inconvénients de ces systèmes de tracé généraux, des packages de visualisation tels que `lattice`  ont été développés ( [Sarkar, 2008](http://lmdvr.r-forge.r-project.org/figures/figures.html) ).
<!-- comment -->

> Ces packages disposent de plusieurs types de graphiques prédéfinis et fournissent une fonction dédiée à la création de chaque type de graphique.

Reviewer: ```suggestion
Ils disposent de plusieurs types de graphiques prédéfinis et fournissent une fonction dédiée à la création de chaque type de graphique.
```

<!-- comment -->

>  Par exemple, `lattice` fournit la fonction `bwplot`  pour créer des diagrammes en boîte et des boîtes à moustaches.

Reviewer: ```suggestion
 Par exemple, `lattice` offre la fonction `bwplot`  pour créer des diagrammes en boîte et des boîtes à moustaches.
```

<!-- comment -->

> L'avantage de ces systèmes est qu'ils facilitent grandement la création de certains graphiques en entiers, y compris une légende et des panneaux.

Reviewer: ```suggestion
L'avantage de ces systèmes est qu'ils facilitent grandement la création de  graphiques complets avec légendes et  panneaux.
```

<!-- comment -->

> L'inconvénient est que l'ensemble des types de graphiques sont prédéfinis, ce qui fait qu'il n'est pas facile de créer des graphiques plus complexes.

Reviewer: ```suggestion
Par contre, ils ont l'inconvénient d'offrir un ensemble de types de graphiques prédéfinis, ce qui fait qu'il n'est pas facile de créer des graphiques plus complexes.
```

<!-- comment -->

Les nouveaux systèmes basés sur la grammaire des graphiques se situent entre ces deux extrêmes.
<!-- comment -->

> Wilkinson a proposé la grammaire des graphiques afin de décrire et de créer une grande classe de graphiques ([Wilkinson, 2005](http://www.springer.com/us/book/9780387245447?wt_mc=GoogleBooks.GoogleBooks.3.EN&token=gbgen#otherversion=9780387286952)).

Reviewer: ```suggestion
En 2005, Leland Wilkinson propose la grammaire des graphiques pour décrire et  créer une grande classe de graphiques ([Wilkinson, 2005](http://www.springer.com/us/book/9780387245447?wt_mc=GoogleBooks.GoogleBooks.3.EN&token=gbgen#otherversion=9780387286952)).
```

knitr::opts_chunk$set(fig.path="Ch02-figures/")
```

Dans ce chapitre nous expliquons la grammaire des graphiques, un modèle puissant pour décrire une vaste gamme de visualisations de données.
<!-- comment -->
Après avoir lu ce chapitre, vous serez en mesure de

<!-- comment -->

* Citer les avantages de la grammaire des graphiques par rapport aux systèmes précédents pour la visualisation de données
<!-- comment -->
* Installer le package R `animint2 `
<!-- comment -->
* Traduire les esquisses de graphique en code ggplot dans R
<!-- comment -->
* Afficher des ggplots sur des pages web en utilisant `animint2`
<!-- comment -->

> * Créer des ggplots avec multiples couches

Reviewer: ```suggestion
* De créer des ggplots multicouches
```

<!-- comment -->

> * Créer des ggplots avec multiples panneaux

Reviewer: ```suggestion
* De créer des ggplots multipanneaux
```

<!-- comment -->
Le package `ggplot2` présente plusieurs avantages par rapport aux systèmes de tracé précédents.

<!-- comment -->

* Comme les systèmes généraux, et contrairement à `lattice`, `ggplot2` impose peu de limites aux types de graphiques qui peuvent être créés (il n'y a pas de types de graphiques prédéfinis).
<!-- comment -->
* Contrairement aux systèmes généraux, et à l'instar de `lattice`, `ggplot2` permet d'inclure facilement des éléments de graphique courants tels que des axes, des panneaux et des légendes.
<!-- comment -->
* Puisque `ggplot2` est basé sur la grammaire des graphiques, il est nécessaire de mapper explicitement les variables de données aux propriétés visuelles. Plus loin dans ce chapitre, nous expliquerons comment ce mappage permet la création d'esquisses qui peuvent être directement traduites en code R.

<!-- comment -->

Enfin, tous les systèmes évoqués précédemment sont destinés à la création de graphiques statiques, qui peuvent être visualisés aussi bien sur un écran d'ordinateur que sur du papier.
<!-- comment -->
Cependant, le sujet principal de ce manuel est `animint2`, un package R pour les graphiques interactifs.
<!-- comment -->

> Contrairement aux graphiques statiques, les graphiques interactifs sont mieux visualisés sur un ordinateur équipé d'une souris et d'un clavier permettant d'interagir avec le graphique.

Reviewer: ```suggestion
Contrairement aux graphiques statiques, les graphiques interactifs sont conçus pour être visualisés sur un ordinateur équipé d'une souris et d'un clavier pour interagir avec le graphique.
```

<!-- comment -->

Étant donné que de nombreux concepts des graphiques statiques sont également utiles dans les graphiques interactifs, le package `animint2` est implémenté en tant qu'extension/fork du package `ggplot2`.
<!-- comment -->

> Dans ce chapitre, nous présenterons les principales caractéristiques de `ggplot2` qui seront également utiles pour la conception de graphiques interactifs dans les chapitres suivants.

Reviewer: ```suggestion
Dans ce chapitre, nous présenterons les principales caractéristiques de `ggplot2` qui serviront seront également dans la conception de graphiques interactifs aux chapitres suivants.
```

> Dans ce chapitre, nous présenterons les principales caractéristiques de `ggplot2` qui seront également utiles pour la conception de graphiques interactifs dans les chapitres suivants.

Reviewer: ```suggestion
Dans ce chapitre, nous présenterons les principales caractéristiques de `ggplot2` utiles également à la conception de graphiques interactifs des chapitres suivants.
```

> Dans ce chapitre, nous présenterons les principales caractéristiques de `ggplot2` qui seront également utiles pour la conception de graphiques interactifs dans les chapitres suivants.

Reviewer: ```suggestion
Dans ce chapitre, nous présenterons les principales caractéristiques de `ggplot2` qui seront également utiles à la conception de graphiques interactifs dans les chapitres suivants.
```

<!-- comment -->

> En 2013, nous avons créé le package `animint`, qui dépend du package `ggplot2`.

Author: ```suggestion
En 2013, Toby Dylan Hocking et ses étudiants ont créé le package `animint`, qui dépend du package `ggplot2`.
```

<!-- comment -->

> Cependant, au cours de la période 2014-2017, le package `ggplot2` a introduit de nombreux changements qui étaient incompatibles avec la grammaire interactive du package `animint`.

Reviewer: ```suggestion
Cependant, au cours de la période 2014-2017, le package `ggplot2` a introduit de nombreux changements incompatibles avec la grammaire interactive du package `animint`.
```

<!-- comment -->

> Par conséquent, en 2018, nous avons créé le package `animint2` qui copie/fork les parties pertinentes du package `ggplot2`.

Reviewer: ```suggestion
ce qui a conduit en 2018, à la création du package `animint2` qui copie/fork les parties pertinentes du package `ggplot2`.
```

<!-- comment -->

> Maintenant, `animint2` peut être utilisé sans que `ggplot2` soit installé.

Reviewer: ```suggestion
`animint2` peut désormais être utilisé sans que `ggplot2` soit installé.
```

<!-- comment -->
En fait, il est conseillé d'utiliser `animint2` sans attacher (avec `library`) `ggplot2`.
<!-- comment -->

> Cependant, il est possible d'utiliser `animint2` avec des packages qui importent/chargent `ggplot2`.

Reviewer: ```suggestion
Il est cependant possible d'utiliser `animint2` avec des packages qui importent/chargent `ggplot2`.
```

<!-- comment -->

> Pour un exemple, voir [Chapitre 16](Ch16-change-point.html) qui utilise le package `penaltyLearning` (qui importe `ggplot2`).

Reviewer: ```suggestion
À titre d'exemple, voir [Chapitre 16](Ch16-change-point.html) qui utilise le package `penaltyLearning` (lequel importe `ggplot2`).
```

<!-- comment -->

## Installer et attacher animint2 {#installing}

<!-- comment -->

> Pour installer la version la plus récente de `animint2` à partir du CRAN,

Reviewer: ```suggestion
Pour installer la dernière version  d' `animint2` à partir du réseau CRAN,
```suggestion
Pour installer la dernière version d' `animint2` à partir du dépôt CRAN,
```
```suggestion
Pour installer la dernière version d' `animint2` depuis le CRAN,
```

<!-- comment -->

> Pour installer une version de développement encore plus récente de `animint2` depuis GitHub,

Reviewer: ```suggestion
Pour installer une version de développement encore plus récente d' `animint2` depuis GitHub,
```

<!-- comment -->

## Traduire des esquisses en ggplots {#sketching}

<!-- comment -->

> Cette section explique comment traduire une esquisse de graphique en code R.

Author: sakjdlalk

<!-- comment -->

## Traduire des esquisses en ggplots {#sketching}

<!-- comment -->

Cette section explique comment traduire une esquisse de graphique en code R.
<!-- comment -->

> Nous utilisons un ensemble de données de la Banque mondiale comme exemple, et nous commençons par charger et examiner un sous-ensemble de ces données.

Reviewer: ```suggestion
Nous utilisons comme exemple un ensemble de données de la Banque mondiale. Commençons par charger et examiner un sous-ensemble de ces données.
```

<!-- comment -->

```{r data}
data(BanqueMondiale, package="animint2fr")
BanqueMondiale$Region <- sub(" (all income levels)", "", BanqueMondiale$région, fixed=TRUE)
head(BanqueMondiale)
tail(BanqueMondiale)
dim(BanqueMondiale)
```

<!-- comment -->

> Le jeu de données `BanqueMondiale` comprend des mesures telles que le taux de fertilité et l'espérance de vie pour chaque pays sur la période 1960-2010.

Reviewer: ```suggestion
Le jeu de données `BanqueMondiale` rassemble des indicateurs tels que le taux de fécondité et l'espérance de vie de chaque pays entre 1960 et 2010.
```

<!-- comment -->

> Le code ci-dessus imprime les premières et dernières lignes, ainsi que la dimension (lignes `r nrow(BanqueMondiale)` et colonnes `r ncol(BanqueMondiale)`) du tableau de données.

Reviewer: ```suggestion
Le code ci-dessus affiche les premières et les dernières lignes, ainsi que les dimensions (lignes `r nrow(BanqueMondiale)` et colonnes `r ncol(BanqueMondiale)`) du tableau de données.

<!-- comment -->

> Supposons que nous voulions voir s'il existe une relation entre l'espérance de vie et le taux de fertilité.

Reviewer: ```suggestion
Supposons que nous voulions voir s'il existe une relation entre l'espérance de vie et le taux de fécondité.
```

<!-- comment -->
 Nous pourrions fixer une année, puis utiliser ces deux variables de données dans un nuage de points.
<!-- comment -->
La figure ci-dessous esquisse les principaux éléments de cette visualisation de données.

<!-- comment -->

![World Bank scatterplot](Ch02-viz-scatter.png)

<!-- comment -->

L'esquisse ci-dessus montre l'espérance de vie sur l'axe horizontal (`x`), le taux de fertilité sur l'axe vertical (`y`) et une légende pour la région.
<!-- comment -->
Ces éléments de l'esquisse peuvent être directement traduits en code R à l'aide de la méthode suivante.
<!-- comment -->

> Tout d'abord, nous devons construire un tableau de données comportant une ligne pour chaque pays en 1975, et des colonnes nommées `espérance.de.vie`, `taux.de.fertilité`, et `région`.

Reviewer: ```suggestion
Tout d'abord, nous devons construire un tableau de données comportant une ligne par pays pour l'année 1975, et des colonnes nommées `espérance.de.vie`, `taux.de.fertilité`, et `région`.
```

<!-- comment -->
Les données `BanqueMondiale` possèdent déjà ces colonnes, il suffit donc de considérer le sous-ensemble pour l'année 1975 :

<!-- comment -->

```{r head}
BanqueMondiale1975 <- subset(BanqueMondiale, année==1975)
head(BanqueMondiale1975)
```

<!-- comment -->

> Le code ci-dessus imprime les données pour 1975, qui ont clairement les colonnes appropriées, et une ligne pour chaque pays

Reviewer: ```suggestion
Le code ci-dessus affiche les données pour 1975, avec les colonnes appropriées et une ligne pour chaque pays
```

<!-- comment -->

> L'étape suivante consiste à utiliser les notes de l'esquisse pour coder un ggplot avec une fonction correspondante de `aes` ou un mappage `aes` des variables de données aux propriétés visuelles :

Author: ```suggestion
L'étape suivante consiste à utiliser les notes de l'esquisse pour coder un ggplot avec `aes` qui fait mappage entre variables et propriétés visuelles :
```

<!-- comment -->

> Si, lorsque vous exécutez le code ci-dessus, l'animint ne s'affiche pas dans votre navigateur web pour une raison quelconque (par exemple si vous voyez une page web vierge), veuillez consulter notre section [FAQ wiki](https://github.com/tdhock/animint2/wiki/FAQ#web-browser-on-local-indexhtml-file-is-blank) qui vous aidera à trouver une solution.

Reviewer: ```suggestion
Lors de l'exécution du code ci-dessus, si l'animint ne s'affiche pas dans votre navigateur web pour une raison quelconque (par exemple si vous voyez une page web vierge), veuillez consulter notre section [FAQ wiki](https://github.com/tdhock/animint2/wiki/FAQ#web-browser-on-local-indexhtml-file-is-blank) qui vous aidera à trouver une solution.
```

<!-- comment -->

> En arrière plan, la fonction `animint` crée une liste avec classes animint, puis R exécute la fonction `print.animint` via la fonction [système d'objets S3](http://adv-r.had.co.nz/OO-essentials.html#s3).

Author: ```suggestion
La fonction `animint()` crée une liste de classe `"animint"`, puis R exécute la fonction `print.animint()` grâce à au [système d'objets S3](http://adv-r.had.co.nz/OO-essentials.html#s3).
```

<!-- comment -->

> Le package `animint2` implémente un compilateur qui prend la liste en entrée et qui produit une page web avec une visualisation de données. 

Reviewer: ```suggestion
Le package `animint2` implémente un compilateur qui prend la liste en entrée et produit une page web avec une visualisation de données. 
```

<!-- comment -->

> Le compilateur est la fonction `animint2dir` qui compile la liste animint `scatter.viz` en un répertoire de fichiers de données et de code qui peuvent être affichés dans un navigateur web.

Author: ```suggestion
Le compilateur est la fonction `animint2dir()` qui compile la liste animint `nuage.vis` en un dossier de fichiers de données (CSV) et de codes (HTML, JavaScript) qui peuvent être affichés dans un navigateur web.
```

<!-- comment -->

> Elle est activée automatiquement par la fonction `print.animint`.

Author: ```suggestion
Le compilateur est invoké par la fonction `print.animint`.
```

<!-- comment -->

> Lorsqu'il est visualisé dans un navigateur web, le graphiquee animint devrait être pratiquement identique aux versions statiques produites par les périphériques graphiques R standard.

Reviewer: ```suggestion
Lorsqu'il est visualisé dans un navigateur web, le graphique animint devrait être pratiquement identique aux versions statiques produites par les périphériques graphiques R standard.
```

<!-- comment -->

> Une différence est que la légende de la région est interactive : en cliquant sur une entrée de la légende, les points de cette couleur sont cachés ou affichés.

Reviewer: ```suggestion
à la différence que la légende de la région est interactive : en cliquant sur une entrée de la légende, les points de cette couleur sont cachés ou affichés.
```

<!-- comment -->

## Visualisation de données multicouches (plusieurs geoms) {#multi-layer}

<!-- comment -->

La visualisation de données multicouches est utile lorsque vous souhaitez afficher plusieurs geoms ou ensembles de données différents dans le même graphique.
<!-- comment -->
Par exemple, considérons l'esquisse suivante qui ajoute une couche de `geom_path` à la visualisation de données précédente.

<!-- comment -->

![multi-layer WorldBank data viz](Ch02-two-layer-viz.png)

<!-- comment -->

Notez que l'esquisse ci-dessus comprend deux geoms différents (point et path).
<!-- comment -->
Les deux geoms partagent une définition commune des éléments `x`, `y`, et de l'`aes` `color`, mais avec des ensembles de données différents.
<!-- comment -->
Nous traduisons ci-dessous cette esquisse en code R.

<!-- comment -->

```{r Ch02-two-layer-viz}
BanqueMondialeAvant1975 <- subset(BanqueMondiale, 1970 <= année & année <= 1975)
deux.couches <- scatter+
  geom_path(aes(
    x=espérance.de.vie, 
    y=taux.de.fertilité, 
    color=région,
    group=pays),
    data=BanqueMondialeAvant1975)
(vis.deux.couches <- animint(deux.couches))
```

<!-- comment -->

> Notez que nous sauvegardons la valeur de retour de la fonction `animint` dans l'objet `vis.deux.couches` (qui est également imprimé en raison des parenthèses).

Reviewer: ```suggestion
Notez que nous sauvegardons la valeur de retour de la fonction `animint` dans l'objet `vis.deux.couches` ( imprimé lui aussi en raison des parenthèses).
```

Author: ```suggestion
Notez que nous sauvegardons la valeur de retour de la fonction `animint()` dans l'objet `vis.deux.couches` (imprimé lui aussi en raison des parenthèses).
```

<!-- comment -->
Dans ce manuel, nous utiliserons souvent des noms de variables commençant par `vis` pour désigner les objets de classe `animint`, qui sont en fait des listes de ggplots et d'options.

<!-- comment -->

> Le graphique ci-dessus montre une visualisation de données avec 2 geoms/couches : 

Reviewer: ```suggestion
Le graphique ci-dessus montre une visualisation de données avec 2 geoms par couche : 
```

<!-- comment -->

* le `geom_point` montre l'espérance de vie, le taux de fertilité et la région de tous les pays en 1975.
<!-- comment -->
* le `geom_path` montre les mêmes variables pour les 5 années précédentes.

<!-- comment -->

L'ajout de `geom_path` montre comment les pays ont changé au fil du temps.
<!-- comment -->

> En particulier, il montre que la plupart des pays se sont déplacés vers la droite et vers le bas, ce qui signifie une espérance de vie plus élevée et un taux de fertilité plus faible.

Reviewer: ```suggestion
Plus particulièrement que les données de la plupart des pays se sont déplacées vers la droite et vers le bas, ce qui signifie une espérance de vie plus élevée et un taux de fécondité plus faible.
```

<!-- comment -->
Il y a cependant quelques exceptions.
<!-- comment -->
Par exemple, les deux pays d'Asie de l'Est situés en bas à gauche ont vu leur espérance de vie diminuer au cours de cette période.
<!-- comment -->

> Par ailleurs, certains pays ont vu leur taux de fertilité augmenter.

Reviewer: ```suggestion
Par ailleurs, certains pays ont vu leur taux de fécondité augmenter.
```

<!-- comment -->

> **Exercice**: essayez de changer la légende `région` par une légende `revenu`.

Reviewer: ```suggestion
**Exercice**: essayez de remplacer la légende `région` par une légende `revenu`.
```

<!-- comment -->
Indice : vous devez utiliser la même spécification `aes(color=revenu)` pour tous les geoms.
<!-- comment -->
Vous pouvez utiliser `scale_color_manual` avec une palette de couleurs séquentielle, voir `RColorBrewer::display.brewer.all(type="seq")` et [lire l'annexe pour plus de détails](Ch99-appendix.html#manual-color-legends).

<!-- comment -->

> Pouvons-nous ajouter les noms des pays à la visualisation de données ?

Reviewer: ```suggestion
Pouvons-nous ajouter le nom des pays à la visualisation de données ?
```

<!-- comment -->
Ci-dessous, nous ajoutons une autre couche avec une étiquette de texte pour le nom de chaque pays.

<!-- comment -->

```{r Ch02-three-layer-viz}
trois.couches <- deux.couches+
  geom_text(aes(
    x=espérance.de.vie, 
    y=taux.de.fertilité, 
    color=région,
    label=pays),
    data=BanqueMondiale1975)
animint(trois.couches)
```

<!-- comment -->

Cette visualisation n'est pas très facile à lire, car il y a beaucoup d'étiquettes de texte qui se chevauchent.
<!-- comment -->
La légende interactive des régions aide un peu, en permettant à l'utilisateur de masquer les données des régions sélectionnées.
<!-- comment -->
Cependant, ce serait encore mieux si l'utilisateur pouvait afficher et masquer le texte pour les pays individuellement.
<!-- comment -->
Ce type d'interaction peut être réalisé en utilisant les mots-clés `showSelected` et `clickSelects` que nous expliquons dans les chapitres 3-4.

<!-- comment -->

Pour l'instant, nous passons à une des grandes forces d'`animint` : la visualisation de données à l'aide de plusieurs graphiques interconnectés.

<!-- comment -->

> ## Visualisation multi-graphiques de données{#multi-plot}

Reviewer: ```suggestion
## Visualisation de données multigraphiques {#multi-plot}
```

Author: ```suggestion
## Visualisations avec plusieurs ggplots interconnectés {#multi-plot}
```

Author: je ne pense pas que "multigraphiques" est assez clair.

<!-- comment -->

> La visualisation de données multi-graphiques est utile lorsque vous souhaitez présenter des ensembles de données liés en utilisant plus d'un mappage `aes`.

Reviewer: ```suggestion
La visualisation de données multigraphiques est utile lorsque vous souhaitez présenter des ensembles de données liés en utilisant plus d'un mappage `aes`.
```

<!-- comment -->
Dans la visualisation de données interactive, un graphique est souvent utilisé pour afficher un sommaire et un autre pour afficher les détails.
<!-- comment -->
Prenons l'exemple d'une visualisation comportant deux graphiques : une série temporelle avec les données de la Banque mondiale de 1960 à 2010 (sommaire) et un nuage de points avec les données de 1975 (détails).
<!-- comment -->
Nous esquissons ci-dessous le graphique des séries temporelles.

<!-- comment -->

![WorldBank data viz with two plots](Ch02-viz-two-plots.png)

<!-- comment -->

Notez que l'esquisse ci-dessus peut être directement traduite dans le code R ci-dessous. 
<!-- comment -->

> Pour plus de simplicité, nous commençons par réduire l'échantillonnage des données à tous les cinq ans (certains explorateurs web comme Chromium ne sont pas en mesure d'afficher plus de 10 000 points de données en une seule fois),

Reviewer: ```suggestion
Pour simplifier, nous réduisons d'abord l'échantillonnage des données à tous les cinq ans (certains explorateurs web comme Chromium ne sont pas en mesure d'afficher plus de 10 000 points de données en une seule fois),
```

<!-- comment -->

```{r}
BanqueMondialeSome <- subset(BanqueMondiale, année %% 5 == 0)
dim(BanqueMondialeSome)
table(BanqueMondialeSome$année)
```

<!-- comment -->

Le tableau de contingence ci-dessus montre qu'il y a 214 lignes pour chacune des années de l'ensemble de données.
<!-- comment -->

> Ensuite, nous copions la visualisation existante (`vis.deux.couches`), puis nous assignons un ggplot à un nouvel élément nommé `timeSeries`. 

Author: ```suggestion
Ensuite, nous copions la visualisation existante (`vis.deux.couches`), puis nous rajoutons un ggplot à un nouvel élément nommé `timeSeries`. 
```

 <!-- comment -->

```{r vizTwoPlots}
vis.deux.graphes <- vis.deux.couches
vis.deux.graphes$timeSeries <- ggplot()+
  geom_line(aes(
    x=année, 
    y=taux.de.fertilité, 
    color=région, 
    group=pays),
    data=BanqueMondialeSome)
```

<!-- comment -->

Il en résulte une liste nommée de deux éléments (les deux éléments sont des ggplots de classe `gganimint`).

<!-- comment -->

```{r summaryTwo}
summary(vis.deux.graphes)
```

<!-- comment -->

> Cette liste de visualisation de données peut être imprimée ou affichée en tapant son nom.

Reviewer: ```suggestion
On peut imprimer ou afficher cette liste de visualisation de données en tapant son nom.
```

<!-- comment -->
Puisque la liste contient deux ggplots, `animint2` affiche la visualisation de données sous la forme de deux graphiques liés.

<!-- comment -->

```{r Ch02-viz-two-plots}
vis.deux.graphes
```

<!-- comment -->

La visualisation de données ci-dessus contient deux ggplots, qui mappent chacun différentes variables de données sur le plan horizontal `x`.
<!-- comment -->
La série temporelle utilise `aes(x=année)` et montre un résumé des valeurs du taux de fertilité pour toutes les années.
<!-- comment -->

> Le nuage de points utilise `aes(x=espérance.de.vie)` et montre les détails de la relation entre le taux de fertilité et l'espérance de vie en 1975.

Reviewer: ```suggestion
Le nuage de points utilise `aes(x=espérance.de.vie)` et montre les détails de la relation entre le taux de fécondité et l'espérance de vie en 1975.
```

<!-- comment -->

**Essayez** de cliquer sur une entrée de légende dans le nuage de points ou dans la série temporelle ci-dessus.
<!-- comment -->
Vous devriez voir les données et les légendes des deux graphiques se mettre à jour simultanément.
<!-- comment -->

> Etant donné que `aes(color=région)` a été spécifié dans les deux graphiques, `animint` crée une seule variable de sélection partagée appelée `région`.

Reviewer: ```suggestion
Étant donné que `aes(color=région)` a été spécifié pour les deux graphiques, `animint` crée une seule variable de sélection partagée appelée `région`.
```

<!-- comment -->

> Cliquer sur l'une des légendes a pour effet de mettre à jour l'ensemble des régions sélectionnées, et `animint` met donc à jour les légendes et les données dans les deux graphiques en conséquence.

Reviewer: ```suggestion
Cliquer sur l'une des légendes a pour effet de mettre à jour l'ensemble des régions sélectionnées,  `animint` met donc à jour les légendes et les données dans les deux graphiques en conséquence.
```

<!-- comment -->

> Il s'agit du mécanisme principal utilisé par `animint` pour créer des visualisations de données interactives avec des graphiques liés, et il sera discuté plus en détail dans les deux prochains chapitres.

Reviewer: ```suggestion
Il s'agit du mécanisme principal utilisé par `animint` pour créer des visualisations de données interactives avec des graphiques liés, nous en discuterons plus en détail dans les deux prochains chapitres.
```

<!-- comment -->

**Exercice**: utilisez `animint` pour créer une visualisation de données avec trois graphiques, en créant une liste avec trois ggplots.
<!-- comment -->
Par exemple, vous pouvez ajouter une série temporelle d'une autre variable de données telle que `espérance.de.vie` ou `population`.

<!-- comment -->

Notez que les deux ggplots mappent la variable taux.de.fertilité sur l'axe y.
<!-- comment -->

> Cependant, comme il s'agit de deux graphiques distincts, les plages de leurs axes y sont calculées séparément.

Reviewer: ```suggestion
Cependant, comme il s'agit de deux graphiques distincts, les plages de leurs axes sont calculées séparément.
```

<!-- comment -->
Cela signifie que même lorsque les deux graphiques sont affichés côte à côte, les deux axes y ne sont pas exactement alignés.
<!-- comment -->

> C'est un problème car il serait plus facile de décoder la visualisation des données si chaque unité d'espace vertical était utilisée pour montrer la même quantité de taux de fertilité.

Reviewer: ```suggestion
C'est problématique, la visualisation des données serait plus facile à décoder si chaque unité d'espace vertical représentait la même quantité de taux de fertilité.
```

Reviewer: Je ne suis pas certaine de quantité de taux de fécondité...ça cloche un peu mais je ne trouve pas par quoi le remplacer.

<!-- comment -->
Pour obtenir cet effet, nous utilisons les facettes dans la section suivante.

<!-- comment -->

> ## Visualisation de données multi-panneaux (facettes) {#multi-panel}

Reviewer: ```suggestion
## Visualisation de données multipanneaux (facettes) {#multi-panel}
```

Author: ```suggestion
## Visualisation de données avec multiples panneaux (facettes) {#facet}
```

<!-- comment -->

Les panneaux ou facettes sont des sous-graphiques qui présentent des visualisations de données connectées.
<!-- comment -->

> L'un des principaux atouts de ggplots est qu'il est relativement facile de créer différents types de graphiques multi-panneaux.

Reviewer: ```suggestion
L'un des principaux atouts de ggplots est qu'il est relativement facile de créer différents types de graphiques multipanneaux.
```

<!-- comment -->

> La visualisation de données multi-panneaux est utile à deux fins différentes :

Reviewer: ```suggestion
La visualisation de données multipanneaux est utile à deux fins différentes :
```

Author: ```suggestion
Les facettes peuvent servir deux objectifs différentes :
```

<!-- comment -->

> * Vous souhaitez aligner les axes de plusieurs graphiques connexes contenant différents geoms. Cela facilite la comparaison entre plusieurs geoms différents et c'est une technique qui est également utile pour la visualisation interactive des données.

Reviewer: ```suggestion
* L'alignement des axes de plusieurs graphiques connexes contenant différents geoms. Cette technique facilite la comparaison entre plusieurs geoms différents et est également utile pour la visualisation interactive des données.
```

> * Vous souhaitez diviser les données d'un geom en plusieurs panneaux. Cette technique facilite la comparaison entre des sous-ensembles de données et est moins utile pour la visualisation interactive de données (l'interactivité peut souvent être utilisée à la place, pour obtenir le même effet de comparaison de sous-ensembles de données).

Reviewer: ```suggestion
* La division des données d'un geom en plusieurs panneaux. Cette technique facilite la comparaison entre des sous-ensembles de données mais elle est moins utile pour la visualisation interactive de données (l'interactivité peut souvent la remplacer et produire le même effet comparatif pour les sous-ensembles de données).
```

<!-- comment -->

> ### Différents geom dans chaque panneau (axes alignés) {#aligned-axes}

Reviewer: ```suggestion
### Des geoms différents pour chaque panneau (axes alignés) {#aligned-axes}
SVP vérifier si cette formulation est plus claire et que le sous-titre de la section suivante lui fait écho
```

<!-- comment -->

> Nous commençons par expliquer comment les facettes sont utiles pour aligner les axes de graphiques apparentés.

Author: ```suggestion
Nous commençons par expliquer comment les facettes sont utiles pour aligner les axes quand on veut afficher des différents geoms dans chaque facette.
```

      x=espérance.de.vie, y=taux.de.fertilité, color=région),
      data=add.x.var(BanqueMondiale1975, "espérance de vie"))+
    geom_path(aes(
      x=espérance.de.vie, y=taux.de.fertilité, color=région,
      group=pays),
      data=add.x.var(BanqueMondialeAvant1975, "espérance de vie"))+
    geom_line(aes(
      x=year, y=taux.de.fertilité, color=région, group=pays),
      data=add.x.var(BanqueMondialeSome, "année"))+
    xlab("")+
    facet_grid(. ~ x.var, scales="free")))
```

<!-- comment -->

La visualisation de données ci-dessus contient un seul ggplot avec deux panneaux et trois couches.
<!-- comment -->
Le panneau de gauche montre les `geom_point` et `geom_path` et le panneau de droite montre les `geom_line`.
<!-- comment -->
Les panneaux ont un axe commun pour le taux de fertilité, ce qui permet de comparer directement les lignes du panneau des séries temporelles avec les points et les chemins du panneau du nuage de points.

<!-- comment -->

Il est à noter que nous avons utilisé la méthode `add.x.var` pour ajouter une variable `x.var` à chaque ensemble de données, puis nous avons utilisé cette variable dans la fonction `facet_grid(scales="free")`.
<!-- comment -->

> Nous appelons cela l'[idiome addColumn then facet](Ch99-appendix.html#addColumn-then-facet) qui est généralement utile pour créer une visualisation de données à plusieurs panneaux avec des axes alignés.

Author: ```suggestion
Nous appelons cette méthode [Ajouter une variable ensuite des facettes](Ch99-appendix.html#addColumn-then-facet) qui est généralement utile pour créer une visualisation de données à plusieurs panneaux avec des axes alignés.
```

<!-- comment -->

> En particulier, si nous voulions changer l'ordre des panneaux dans la visualisation, il nous suffirait de modifier l'ordre des niveaux de facteurs dans la définition de `add.x.var` .

Reviewer: ```suggestion
Ainsi, pour changer l'ordre des panneaux dans la visualisation, il nous suffirait de modifier l'ordre des niveaux de facteurs dans la définition de `add.x.var` .
```

<!-- comment -->

> Notez également que `theme_bw` signifie qu'il faut utiliser des bordures de panneaux noires et des fonds de panneaux blancs, et que `panel.margin=0` signifie qu'il n'y a pas d'espace entre les panneaux.

Author: ```suggestion
Notez également que `theme_bw()` fait les bordures de panneaux noires et des fonds de panneaux blancs, et que `theme(panel.margin=0)` pourrait être utilisé pour supprimer l'espacement entre les panneaux.
```

<!-- comment -->

> En éliminant l'espace entre les panneaux, on utilise plus d'espace pour les panneaux, ce qui permet d'accentuer les données.

Reviewer: ```suggestion
Cette élimination libère de l'espace pour les panneaux, ce qui permet de mettre l'accent sur les données.
```

<!-- comment -->

> C'est ce que nous appelons l'[idiome facettes économes en espace](Ch99-appendix.html#space-saving-facets) qui est généralement utile dans tout ggplot avec des facettes.

Reviewer: ```suggestion
C'est ce que nous appelons l'[idiome facettes économes en espace](Ch99-appendix.html#space-saving-facets), il est généralement utile dans tout ggplot avec des facettes.
```

<!-- comment -->

Dans la visualisation ci-dessus, les étiquettes de texte se chevauchent un peu, ce qui peut être corrigé par l'une des méthodes suivantes (exercice pour le lecteur).

<!-- comment -->

* en utilisant l'argument `breaks` de `scale_x_continuous()`
<!-- comment -->
* en réduisant la taille du texte avec `theme()` et `element_text()`
<!-- comment -->
* en augmentant la largeur du graphique à l'aide de `theme_animint()` voir [Chapitre 6](Ch06-other.html#plot-height-width) pour plus d'informations.

<!-- comment -->

> ### Mêmes geoms dans chaque panneau (réalisation de sous-ensembles de données) {#compare-data-subsets}

Author: ```suggestion
### Facettes pour comparaison des sous-ensembles de données {#compare-data-subsets}
```

<!-- comment -->

La deuxième raison d'utiliser des graphes avec plusieurs panneaux dans une visualisation de données est de comparer des sous-ensembles d'observations.
<!-- comment -->

> Cela facilite la comparaison entre les sous-ensembles de données et peut être utilisé dans au moins deux situations différentes :

Reviewer: ```suggestion
Cette approche facilite la comparaison entre les sous-ensembles de données et peut être utilisée dans au moins deux situations différentes :
```

<!-- comment -->

> * L'ensemble de données d'un geom comporte trop d'observations pour être affiché de manière informative dans un seul panneau.

Reviewer: ```suggestion
* L'ensemble de données d'un geom comporte trop d'observations pour être visualisé efficacement dans un seul panneau.
```

<!-- comment -->

Prenons l'exemple de l'esquisse ci-dessous.

<!-- comment -->

![World Bank panels plot](Ch02-viz-panels.png)

<!-- comment -->

Notez que les trois panneaux tracent les deux mêmes geoms (point et path).
<!-- comment -->
Puisque `facet=show.année` et qu'il y a trois panneaux, nous devrons créer des tableaux de données qui auront trois valeurs pour la variable `show.année`.
<!-- comment -->
Le `geom_point` ne contient des données que pour 3 ans, et le tableau `geom_path` a des données pour 15 ans (mais 3 valeurs de `show.année`).
<!-- comment -->
Le code ci-dessous crée ces deux ensembles de données pour trois années de l'ensemble de données de la Banque mondiale.

<!-- comment -->

```{r}
show.point.list <- list()
show.path.list <- list()
for(show.année in c(1975, 1985, 1995)){
  show.point.list[[paste(show.année)]] <- data.frame(
    show.année, subset(BanqueMondiale, année==show.année))
  show.path.list[[paste(show.année)]] <- data.frame(
    show.année, subset(BanqueMondiale, show.année - 5 <= année & année <= show.année))
}
show.point <- do.call(rbind, show.point.list)
show.path <- do.call(rbind, show.path.list)
```

<!-- comment -->

> Nous avons utilisé une boucle for sur trois valeurs de `show.année` la variable que nous utiliserons plus tard dans `facet_grid`.

Reviewer: ```suggestion
Nous avons utilisé une boucle for sur trois valeurs de `show.année`, la variable que nous utiliserons plus tard dans `facet_grid`.
```

<!-- comment -->

Ci-dessous, nous appliquons les facettes sur la variable `show.année` pour créer une visualisation de données avec trois panneaux.

<!-- comment -->

```{r Ch02-viz-panels}
animint(
  scatter=ggplot()+
    geom_point(aes(
      x=espérance.de.vie, y=taux.de.fertilité, color=région),
      data=show.point)+
    geom_path(aes(
      x=espérance.de.vie, y=taux.de.fertilité, color=région,
      group=pays),
      data=show.path)+
    facet_grid(. ~ show.année)+
    theme_bw())
```

<!-- comment -->

La visualisation de données ci-dessus contient un seul ggplot avec trois panneaux
<!-- comment -->
Elle montre une plus grande partie de l'ensemble des données de la Banque mondiale que les visualisations précédentes qui ne montraient que les données de 1975.
<!-- comment -->
Cependant, elle ne montre qu'un sous-ensemble de données relativement restreint.
<!-- comment -->
Vous pourriez être tenté d'utiliser un panneau pour afficher chaque année (pas seulement 1975, 1985 et 1995).
<!-- comment -->
Sachez toutefois que ce type de visualisation de données à plusieurs panneaux n'est utile que s'il n'y a que quelques sous-ensembles de données.
<!-- comment -->

> Au-delà d'une dizaine de panneaux, il devient difficile de voir toutes les données en même temps, et donc de faire des comparaisons significatives.

Reviewer: ```suggestion
Au-delà d'une dizaine de panneaux, il devient difficile de voir toutes les données simultanément, et donc de faire des comparaisons significatives.
```

<!-- comment -->

> Au lieu de montrer toutes les données en même temps, nous pouvons créer une visualisation de données animée qui montre à l'observateur différents sous-ensembles de données au fil du temps.

Reviewer: ```suggestion
Au lieu de montrer toutes les données à la fois, nous pouvons créer une visualisation de données animée qui présente à l'observateur différents sous-ensembles de données au fil du temps.
```

<!-- comment -->
Dans le [chapitre suivant](Ch03-showSelected.html) nous montrerons comment le nouveau mot-clé `showSelected` peut être utilisé pour réaliser une animation, et nous révélerons plus de détails sur cet ensemble de données.

<!-- comment -->

## Résumé du chapitre et exercices {#exercises}

<!-- comment -->

Ce chapitre a présenté les bases de la visualisation de données statiques avec `ggplot2`.
<!-- comment -->
Nous avons montré comment animint peut être utilisé pour afficher une liste de ggplots dans un navigateur web.
<!-- comment -->
Nous avons expliqué deux caractéristiques de `ggplot2` qui le rendent idéal pour la visualisation de données : les graphiques avec multiples couches et multi-panneaux.

<!-- comment -->

Exercices :

<!-- comment -->

* Quels sont les trois principaux avantages de `ggplot2` par rapport aux systèmes antérieurs tels que `grid` et `lattice` ?
<!-- comment -->

> * Quel est l'objectif des graphiques multi-couches ?

Author: ```suggestion
* Pourquoi on utiliserait plusieurs couches dans un même graphique ?
```

<!-- comment -->

> * Quelles sont les deux raisons différentes de créer des graphiques à panneaux multiples ? Lequel de ces deux types est utile pour l'interactivité ?

Reviewer: ```suggestion
* Quelles sont les deux raisons pour créer des graphiques multipanneaux ? Lequel de ces deux types est utile pour l'interactivité ?
```

<!-- comment -->

**Exercice**: utilisez `animint` pour créer une visualisation de données avec trois graphiques, en créant une liste avec trois ggplots.
<!-- comment -->
Par exemple, vous pouvez ajouter une série temporelle d'une autre variable de données telle que `espérance.de.vie` ou `population`.

<!-- comment -->

Notez que les deux ggplots mappent la variable taux.de.fertilité sur l'axe y.
<!-- comment -->
Cependant, comme il s'agit de deux graphiques distincts, les plages de leurs axes y sont calculées séparément.
<!-- comment -->
Cela signifie que même lorsque les deux graphiques sont affichés côte à côte, les deux axes y ne sont pas exactement alignés.
<!-- comment -->
C'est problématique, la visualisation des données serait plus facile à décoder si chaque unité d'espace vertical représentait la même quantité de taux de fertilité.
<!-- comment -->
Pour obtenir cet effet, nous utilisons les facettes dans la section suivante.

<!-- comment -->

## Visualisation de données multi-panneaux (facettes) {#multi-panel}

<!-- comment -->

> Les panneaux ou facettes sont des sous-graphiques qui présentent des visualisations de données connectées.

Author: ```suggestion
Un ggplot peut avoir des facettes, qui sont des sous-graphiques, chacun qui présent un différent sous-ensemble de données.
```

> Un ggplot peut avoir des facettes (panneaux), qui sont des sous-graphiques, chacun qui présent un différent sous-ensemble de données.

Reviewer: ```suggestion
Un ggplot peut avoir des facettes (panneaux), qui sont des sous-graphiques, chacun d'eux présentant un sous-ensemble de données différent.
```

<!-- comment -->

Il est à noter que nous avons utilisé la méthode `add.x.var` pour ajouter une variable `x.var` à chaque ensemble de données, puis nous avons utilisé cette variable dans la fonction `facet_grid(scales="free")`.
<!-- comment -->
Nous appelons cette méthode [Ajouter une variable ensuite des facettes](Ch99-appendix.html#addColumn-then-facet). Elle est généralement utile pour créer une visualisation de données à plusieurs panneaux avec des axes alignés.
<!-- comment -->
Ainsi, pour changer l'ordre des panneaux dans la visualisation, il nous suffirait de modifier l'ordre des niveaux de facteurs dans la définition de `add.x.var` .

<!-- comment -->

Notez également que `theme_bw()` fait les bordures de panneaux noires et des fonds de panneaux blancs, et que `theme(panel.margin=0)` pourrait être utilisé pour supprimer l'espacement entre les panneaux.
<!-- comment -->
Cette élimination libère de l'espace pour les panneaux, ce qui permet de mettre l'accent sur les données.
<!-- comment -->

> Nous appelons cette méthode [Les facettes économes en espace](Ch99-appendix.html#space-saving-facets) qui est généralement utile dans tout ggplot avec des facettes.

Reviewer: ```suggestion
Nous appelons cette méthode [Les facettes économes en espace](Ch99-appendix.html#space-saving-facets), elle est généralement utile dans tout ggplot avec des facettes.
```

	* ggplot < layer
  <!-- comment -->
	
* Dans les facettes `vis.alignée`, pourquoi est-il important d'utiliser l'argument `scales="free"` ?
<!-- comment -->
* Dans `vis.alignée` nous avons montré un ggplot avec un panneau de nuage de points à gauche et un panneau de séries temporelles à droite. Réalisez une autre version de la visualisation de données avec le panneau des séries temporelles à gauche et le panneau du nuage de points à droite.
<!-- comment -->
* Dans `vis.alignée` le nuage de points affiche le taux de fécondité et l'espérance de vie, mais la série chronologique n'affiche que le taux de fertilité. Réalisez une autre version de la visualisation de données qui montre les deux séries temporelles. Indice : utilisez les panneaux horizontaux et verticaux dans `facet_grid`.
<!-- comment -->
* Utilisez `aes(size=population)` dans le nuage de points pour indiquer la population de chaque pays. Indice : `scale_size_animint(pixel.range=c(5, 10))` signifie que des cercles d'un rayon de 5/10 pixels doivent être utilisés pour représenter la population minimale/maximale.
<!-- comment -->

> * Créez une visualisation de données à plusieurs panneaux qui montre chaque année de la `BanqueMondiale` dans un panneau séparé. Quelles sont les limites de l'utilisation de graphiques statiques pour visualiser ces données ?

Author: ```suggestion
* Créez une visualisation de données à plusieurs panneaux qui montre chaque année des données `BanqueMondiale` dans un différent panneau. Quelles sont les limites de l'utilisation de graphiques statiques pour visualiser ces données ?
```

<!-- comment -->
* Créez un `vis.alignée` en utilisant un système de tracé qui n'est pas basé sur la grammaire des graphiques. Par exemple, vous pouvez utiliser les fonctions du package `graphics` dans R (`plot`,`points`,`lines` etc.), ou `matplotlib` en Python. Quels sont les avantages de `ggplot2` et `animint` ?

<!-- comment -->

> Ensuite, [Chapitre 3](Ch03-showSelected.html) explique le mot-clé `showSelected` qui indique une variable à utiliser pour le sous-ensemble des données avant le tracé.

Reviewer: ```suggestion
Dans le [Chapitre 3](Ch03-showSelected.html) qui suit, nous expliquons le mot-clé `showSelected` qui indique une variable à utiliser pour le sous-ensemble des données avant le tracé.
```

Reviewer: Je ne suis pas certaine du segment "qui indique une variable à utiliser pour le sous-ensemble des données avant le tracé". Pourriez-vous vérifier? Il me semble que ca ne correspond pas à la définition de showselect du début du chapitre 3.
